src/internal/storage/fileset/buffer.go

```diff
@@ -25,52 +33,75 @@ func NewBuffer() *Buffer {
 	}
 }
 
-func (b *Buffer) Add(path, datum string) io.Writer {
+func (b *Buffer) addInternal(path, datum string) *file {
 	path = Clean(path, false)
 	if _, ok := b.additive[path]; !ok {
 		b.additive[path] = make(map[string]*file)
 	}
 	datumFiles := b.additive[path]
 	if _, ok := datumFiles[datum]; !ok {
+		b.fileCount++
```

Comment:
This is super nitpicky, but I generally prefer the primary functionality (creating the datum file) to come before side effects (updating the file count). It just reads better imo.

Reply:
makes sense, I was also debating just running through everything to re-count every once in a while rather than the incremental bookkeeping